tensorboard/webapp/widgets/histogram/histogram_card_fob_controller.ts

```diff
@@ -24,30 +30,50 @@ import {TemporalScale} from './histogram_component';
     <card-fob-controller
       [axisDirection]="axisDirection"
       [timeSelection]="timeSelection"
-      [cardAdapter]="this"
+      [axisPositionFromStartStep]="getAxisPositionFromStartStep()"
+      [axisPositionFromEndStep]="getAxisPositionFromEndStep()"
+      [highestStep]="getHighestStep()"
+      [lowestStep]="getLowestStep()"
+      [cardFobHelper]="cardFobHelper"
       (onTimeSelectionChanged)="onTimeSelectionChanged.emit($event)"
       (onTimeSelectionToggled)="onTimeSelectionToggled.emit()"
     ></card-fob-controller>
   `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class HistogramCardFobController implements CardFobAdapter {
+export class HistogramCardFobController {
   @Input() steps!: number[];
   @Input() timeSelection!: TimeSelection;
   @Input() temporalScale!: TemporalScale;
   @Output() onTimeSelectionChanged = new EventEmitter<TimeSelection>();
   @Output() onTimeSelectionToggled = new EventEmitter();
 
   readonly axisDirection = AxisDirection.VERTICAL;
+  readonly cardFobHelper: CardFobGetStepFromPositionHelper = {
+    getStepHigherThanAxisPosition:
+      this.getStepHigherThanAxisPosition.bind(this),
+    getStepLowerThanAxisPosition: this.getStepLowerThanAxisPosition.bind(this),
+  };
+
+  getAxisPositionFromStartStep() {
+    return this.temporalScale(this.timeSelection.start.step);
+  }
+
+  getAxisPositionFromEndStep() {
```

Comment:
Note:

@Input() axisPositionFromEndStep!: number;

... does not include `null` in the type. Unfortunately our Angular compiler does not catch this in the OSS code base but we should consider updating the return type to better reflect the possibility.